check_api/src/main/java/com/google/errorprone/bugpatterns/BugChecker.java

```diff
@@ -269,8 +270,12 @@ public boolean isSuppressed(Symbol symbol) {
   }
 
   private boolean isSuppressed(SuppressWarnings suppression) {
-    return suppression != null
-        && !Collections.disjoint(Arrays.asList(suppression.value()), allNames());
+    if (suppression == null || !supportsSuppressWarnings()) {
+      return false;
+    }
+
+    List<String> suppressions = Arrays.asList(suppression.value());
+    return suppressions.contains("all") || !Collections.disjoint(suppressions, allNames());
   }
```

Comment:
This method is effectively deprecated, since it's only called by deprecated code; the first test below covers it.